module/module.go

```diff
@@ -336,7 +336,7 @@ func OpenOrCreateModule(modulePath string, url string) Module {
 
 	log.Debug("Module directory does not exists.\n")
 
-	if strings.HasSuffix(url, ".git") {
+	if strings.HasSuffix(url, ".git") || moduleType == "git" {
```

Comment:
if moduleType == tar.gz, but the url has a git suffix, we would still create a git module. I think the moduleType should have precedence if set. Let's have a `determineModuleType` function that takes the moduleType and url and returns an enum?